Linux requires the Launcher to have the `CAP_SYS_RESOURCE` capability inorder to run

**Describe the bug**
On startup of the GameLauncher/ServerLauncher applications in O3DE, it tries to update the [ulimits](https://github.com/o3de/o3de/blob/development/Code/LauncherUnified/Platform/Common/UnixLike/Launcher_UnixLike.cpp#L58) of the process itself by increasing the value of the Maximum Core Size file and increasing the Process Stack Size.

If the launcher doesn't have the `CAP_SYS_RESOURCE`, and their resource "hard limit' values for the Maximum Core Size file is less than infinity and the Process Stack Size is than the max, then when the launcher invokes [setrlimit](https://linux.die.net/man/2/setrlimit)
 it fails.

This results in the GameLauncher [immediately exiting](https://github.com/o3de/o3de/blob/development/Code/LauncherUnified/Launcher.cpp#L446-L450.
)

**Steps to reproduce**
1. Build the GameLauncher on Linux.
```bash
cmake --build <build-dir> --target ${project}.GameLauncher --config profile
```
2. Check that the GameLauncher doesn't have the `CAP_SYS_RESOURCE` capability using the `getcap` command.
3. Attempt to run the GameLauncher on Linux with an unprivileged user account.
4. The GameLauncher will immediately exit and the Game.log/terminal window will contain "Error: Operation not permitted" error

**Expected behavior**
GameLauncher launches without exiting due to permission error

**Actual behavior**
GameLauncher fails to launch with error.
`[ERROR] Failed to update resource limit for resource 4.  Error: Operation not permitted`

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: [] 
 - OS: [Debian GNU/ Fedora]
 - CPU []
 - GPU []
 - Memory []

**Additional context**
What we should do is to remove the `setrlimit` calls on Linux inside of the Launcher_UnixLike.cpp and have that function just return true.
If there is a need for the process capabilities to be changed, we should have the user do it on their machine.

--- a/Code/LauncherUnified/Platform/Common/UnixLike/Launcher_UnixLike.cpp
+++ b/Code/LauncherUnified/Platform/Common/UnixLike/Launcher_UnixLike.cpp
@@ -49,18 +49,17 @@ namespace
         rlimit limit;
         if (getrlimit(resource, &limit) != 0)
         {
-            AZ_Error("Launcher", false, "[ERROR] Failed to get limit for resource %d.  Error: %s", resource, strerror(errno));
-            return false;
+            AZ_Warning("Launcher", false, "[WARNING] Unable to get limit for resource %d.  Error: %s", resource, strerror(errno));
         }
 
         if (updateLimit(limit))
         {
             if (setrlimit(resource, &limit) != 0)
             {
-                AZ_Error("Launcher", false, "[ERROR] Failed to update resource limit for resource %d.  Error: %s", resource, strerror(errno));
-                return false;
+                AZ_Warning("Launcher", false, "[WARNING] Unable to update resource limit for resource %d.  Error: %s", resource, strerror(errno));
             }
         }
+
         return true;
     }
 }

